flow_judge/utils/result_writer.py

```diff
@@ -28,24 +28,35 @@ def write_results_to_disk(
     }
 
     metric_folder = os.path.join(output_dir, fmt_metric_name)
-    os.makedirs(metric_folder, exist_ok=True)
+    metadata_folder = os.path.join(metric_folder, f"metadata_{fmt_metric_name}_{fmt_model_id}")
+
+    # Create all necessary directories
+    os.makedirs(metadata_folder, exist_ok=True)
 
     base_filename = f"{fmt_metric_name}_{fmt_model_id}_{model_metadata['model_type']}_{timestamp}"
-    metadata_path = os.path.join(metric_folder, f"metadata_{base_filename}.jsonl")
+    metadata_path = os.path.join(metadata_folder, f"metadata_{base_filename}.jsonl")
```

Comment:
You might have not pushed this change, still jsonl

Reply:
In the newest we have the metadata as json and results as jsonl.
https://github.com/flowaicom/flow-judge/blob/91937b8ca7708d6cc534570065a7b6e7137d87ea/flow_judge/utils/result_writer.py#L37